File: service/src/test/java/com/example/GreetingServiceTest.java
package com.example;

import org.junit.Before;
import org.junit.Test;

import static com.example.PersonFixture.jim;
import static org.hamcrest.CoreMatchers.equalTo;
import static org.junit.Assert.assertThat;

public class GreetingServiceTest {

    private GreetingService service;

    @Before
    public void setUp(){
        service = new GreetingService();
    }

    @Test
    public void testGreeting(){
        assertThat(service.greet(jim()), equalTo("Hello Jim"));
    }
}

File: service/src/test/kotlin/com/example/GreetingServiceTest.kt
package com.example

import org.junit.Before
import org.junit.Test

import org.hamcrest.CoreMatchers.equalTo
import org.junit.Assert.assertThat

class GreetingServiceTest {

    private lateinit var service: GreetingService

    @Before
    fun setUp() {
        service = GreetingService()
    }

    @Test
    fun testGreeting() {
        assertThat<String>(service.greet(jim()), equalTo<String>("Hello Jim"))
    }
}
